scipy/stats/_survival.py

```diff
@@ -0,0 +1,116 @@
+from dataclasses import make_dataclass
+import numpy as np
+from ._censored_data import CensoredData
+
+
+__all__ = ['ecdf']
+
+
+ECDFResult = make_dataclass('ECDFResult', ['x', 'cdf', 'sf'])
+
+
+def ecdf(sample):
+    """Empirical cumulative distribution function of a sample
+
+    The empirical cumulative distribution function (ECDF) is a step function
+    estimate of the CDF of the distribution underlying a sample.
+
+    Parameters
+    ----------
+    sample : 1D array_like or `stats.CensoredData`
+        Besides array_like, instances of `stats.CensoredData` containing
+        uncensored observations are supported. Currently, instances of
+        `stats.CensoredData` with censored data will result in a
+        ``NotImplementedError``, but future support for left-censored,
+        right-centered, and interval-censored data is planned.
+
+    Returns
+    -------
+    An object with the following attributes.
+
+    x : ndarray
+        The unique values at which the ECDF changes.
+    cdf : ndarray
+        The values of the ECDF corresponding with `x`.
+    sf : ndarray
+        The empirical survival function, the complement of the ECDF.
```

Comment:
Is this really needed if in the end we don't do more than `1-CDF`?

Reply:
When we add support for CensoredData, many people will want to use this function to get the SF. ("survival analysis")

Reply:
Ok so conveniency. Fine with me 👍